nasa/harmony-service-lib-py
# HARMONY-1862: Update pystac library dependency

## Jira Issue ID
HARMONY-1862

## Description
This updates `pystac` compatibility by allowing you to use newer versions of `pystac`. This should allow you to use the latest version of `pystac` if using python 3.9 or 3.10.

## Local Test Steps

Test a service (with local Harmony deployment):
1. Checkout branch HARMONY-1862 of this repo
2. If testing with harmony service example, also checkout branch HARMONY-1862 of the harmony service example repo
3. Build a service image using this branch of the service lib (see the service example [README](https://github.com/nasa/harmony-service-example/blob/HARMONY-1862/README.md) for instructions on building a service with a local copy/branch of the service lib). 
4. Update your local Harmony instance service deployment to use the newly built image.
5. Run an example request and make sure it succeeds.

Standalone test:
1. Activate your virtual environment (python 3.9 or 3.10).
2. Run `make install`
3. Specify a newer version of pystac in your virtual environment (e.g. `pip install pystac==1.10.1`).
4. Run `make test`.

## PR Acceptance Checklist
